### DISCUSSION TITLE
Print("hello")

### DISCUSSION THREAD

**Discussion by IlanVinograd** [carbon-language/carbon-lang#4653] [Getting started] [answered]
I tired print string by using docker to build the code on windows but I'm getting error
```
fn Run() {
  Core.Print("Hello");
}
```

I used dockerfile:
```
# Use Ubuntu 22.04 as the base image
FROM ubuntu:22.04

# Set environment variables to suppress interactive prompts
ENV DEBIAN_FRONTEND=noninteractive

# Install necessary tools and dependencies
RUN apt-get update && apt-get install -y \
    wget \
    gcc \
    g++ \
    libgcc-11-dev \
    tar \
    && apt-get clean

# Set the working directory inside the container
WORKDIR /carbon-lang

# Copy the local files to the container
COPY . .

# Make the script executable
RUN chmod +x script.sh

# Default command to run the script
CMD ["./script.sh"]
```

and script.sh:

```
#!/bin/bash

# Get the nightly version from yesterday
VERSION="$(date -d yesterday +0.0.0-0.nightly.%Y.%m.%d)"

# Download the Carbon toolchain
wget https://github.com/carbon-language/carbon-lang/releases/download/v${VERSION}/carbon_toolchain-${VERSION}.tar.gz

# Unpack the toolchain
tar -xvf carbon_toolchain-${VERSION}.tar.gz

# Compile the renamed Carbon source file
./carbon_toolchain-${VERSION}/bin/carbon compile --output=main.o main.carbon

# Link to an executable
./carbon_toolchain-${VERSION}/bin/carbon link --output=main main.o

# Run the compiled executable
./main
```

**Comment by jonmeow** [comment, marked as answer]
Sorry, but printing strings is not supported right now.

Carbon is still in early development. Right now, it can compile code, but we are still building features to make it usable. The [sieve example](https://github.com/carbon-language/carbon-lang/blob/trunk/examples/sieve.carbon) shows where things are.

The main evaluation version will be the 0.1 milestone, which is [on the roadmap for 2025](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/roadmap.md#potential-2025-goals-ship-a-working-01-language-for-evaluation). I expect it'll start being more usable over the next year, but building the full [0.1 feature list](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/milestones.md#language-features) would optimistically be late 2025.

We're happy to have the interest, just a lot of things missing still.
